Implement autostart on linux

Hofer-Julian opened the pull request:
Implements autostart on linux using [XDG Autostart](https://specifications.freedesktop.org/autostart-spec/autostart-spec-latest.html), which should be supported by all [popular desktop environments](https://wiki.archlinux.org/index.php/Autostarting)
Icon is currently not displayed when entry is viewed with gnome tweaks.
We would probably have to insert the absolute path to vorta's icon for that.

m3nu commented on src/vorta/utils.py:
```diff
@@ -245,6 +248,25 @@ def open_app_at_startup(enabled=True):
                                                  None, None, url, props, None)
         if not enabled:
             LSSharedFileListItemRemove(login_items, new_item)
+    elif sys.platform == 'linux':
```
Will this also work on non-Gnome desktops and window managers? For Linux there is more fragmentation, so I'm generally skeptical.

Hofer-Julian replied:
I just looked it up, and apparently it is recommended to use 'sys.platform.startswith('linux')' instead to maintain compatibility for python versions <3.3. I changed that now.

Otherwise it should work, but maybe @ThomasWaldmann and @antlarr can help to test it on xfce and KDE?

m3nu commented:
Implementation looks good. I rebased your PR to get ride of the (unrelated) Travis errors and moved all autostart-related functions to a new file.

If it still works for you, it's good to merge.

ThomasWaldmann commented on src/vorta/autostart.py:
```diff
@@ -0,0 +1,71 @@
+from setuptools.command.install import install
+
+
+GNOME_STARTUP_FILE = """[Desktop Entry]
```
style tip:

foo = """\
first line
second
...
"""

Hofer-Julian replied:
Nice, I will remember that one!

m3nu replied:
Changed. Good tip.

Hofer-Julian commented:
> Implementation looks good. I rebased your PR to get ride of the (unrelated) Travis errors and moved all autostart-related functions to a new file.
> 
> If it still works for you, it's good to merge.

Your changes look good, the code is far more readable now.

I wanted to try something with git/github and I deleted my fork in the process, and now I seem to not be able to commit to this pull request anymore.

So here would be my changes.

1. Remove quotation mark from "Exec={}/vorta in line 11
2. Rename GNOME_STARTUP_FILE to LINUX_STARTUP_FILE and remove reference to Gnome in the comment, because it should work on (nearly) all linux distributions

m3nu commented:
Oops. Just don't delete the branch on Github or I'll have trouble merging it. Will apply your changes.

Hofer-Julian commented:
> Oops. Just don't delete the branch on Github or I'll have trouble merging it. Will apply your changes.

Yeah, I'll reduce explorative learning to a minimum now 🙈

m3nu commented:
No damage done. It's OK to break stuff once in a while.

I'm closing this in lieu of #191, since it can't merge from a deleted branch.
